torch/csrc/jit/type.h

```diff
@@ -733,21 +775,43 @@ struct VarType : public Type {
   std::string name_;
 };
 
+// used in FunctionSchema to keep track of alias information
+// value->type() should never contain this kind of type.
+struct AnnotatedType;
```

Comment:
Discussed in person: this will get replaced with a side structure that follows the `containedTypes()` layout of the original type but adds alias annotations.